omod/src/main/java/org/openmrs/web/servlet/LoginServlet.java

```diff
@@ -178,12 +180,24 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 			catch (ContextAuthenticationException e) {
 				// set the error message for the user telling them
 				// to try again
-				httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR, "auth.password.invalid");
+				Integer maximumAlowedAttempts = 7;
+				if (loginAttemptsByUser <= maximumAlowedAttempts) {
+					Integer remainingAttempts = maximumAlowedAttempts - loginAttemptsByUser;
+					httpSession.setAttribute(WebConstants.OPENMRS_ERROR_ATTR,
```

Comment:
What happens for users who are using other non English locales?

Reply:
well, reason i ddnt  want to use the message.properties, i thought they arent detailed enough